### PR TITLE
Add s3caller worker config validation and tests

### PR DESCRIPTION
**Pull request by nvinuesa**
The manifold config of the s3caller worker was not being validated, add validation.

Also, while adding tests, some cleanup was done on the s3caller manifold:
- remove AgentName, which is not used
- remove Filter, which is not used

## Checklist

- [X] Code style: imports ordered, good names, simple structure, etc
- [X] Comments saying why design decisions were made
- [X] Go unit tests, with comments saying what you're testing
- [ ] ~[Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- [ ] ~[doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

```
go test github.com/juju/juju/worker/s3caller/... -gocheck.v   
```
also
```
juju bootstrap localhost c
juju add-model m
juju deploy ubuntu
```

## Links

**Jira card:** JUJU-4724

### CODE REVIEW

File: worker/deployer/unit_manifolds.go [hunk 1]
```diff
@@ -135,11 +135,9 @@ func UnitManifolds(config UnitManifoldsConfig) dependency.Manifolds {
 		// API for uploading and downloading charms. It provides a
 		// S3-compatible API.
 		s3CallerName: s3caller.Manifold(s3caller.ManifoldConfig{
-			AgentName:            agentName,
 			APIConfigWatcherName: apiConfigWatcherName,
 			APICallerName:        apiCallerName,
 			NewS3Client:          s3caller.NewS3Client,
-			Filter:               connectFilter,
```
Comment (SimonRichardson) [inline worker/deployer/unit_manifolds.go:142]: I think this is actually required.

**Review by SimonRichardson — CHANGES_REQUESTED** [review]

File: worker/s3caller/package_test.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+package s3caller
+
+import (
+	"testing"
+
+	"go.uber.org/mock/gomock"
+	gc "gopkg.in/check.v1"
+
+	jujujujutesting "github.com/juju/juju/testing"
```
Comment (jack-w-shaw) [inline worker/s3caller/package_test.go:12]: Probably best to rename this `coretesting`, as it is called in a few other places

File: worker/deployer/unit_manifolds.go (hunk 1 above)
Comment (nvinuesa) [reply worker/deployer/unit_manifolds.go:142]: but it's never used on the `s3caller`, why is it required?

File: worker/s3caller/package_test.go (hunk 2 above)
Comment (nvinuesa) [reply worker/s3caller/package_test.go]: done!

**Review by SimonRichardson — APPROVED** [review]

**Pull request merged**